File: 通用代码/门面模式/Facade.java
//模块一可以完整的访问所有业务逻辑
public class Facade{
    //被委托的对象
    private ClassA a = new ClassA();
    private ClassB b = new ClassB();
    private ClassC c = new ClassC();
    private Context context = new Context();
    //提供给外部访问的方法
    public void methodA(){
        this.a.doSomethingA();
    }
    public void methodB(){
        this.b.doSomethingB();
    }
    public void methodC(){
        this.c.doSomethingC();
    }
    public void methodD(){
        this.context.complexMethod();
    }
}

File: 通用代码/门面模式(C++)/Facade.cpp
#include "Facade.h"

Facade::Facade()
{
    this->_a = new ClassA();
    this->_b = new ClassB();
    this->_c = new ClassC();
    this->_context = new Context();
}

Facade::~Facade()
{
    delete this->_a;
    delete this->_b;
    delete this->_c;
    delete this->_context;

    this->_a = NULL;
    this->_b = NULL;
    this->_c = NULL;
    this->_context = NULL;
}

void Facade::methodA(){
    this->_a->doSomethingA();
}
void Facade::methodB(){
    this->_b->doSomethingB();
}
void Facade::methodC(){
    this->_c->doSomethingC();
}
void Facade::methodD(){
    this->_context->complexMethod();
}
